### ISSUE TITLE
Gatsby does not support new JSX transform

### ISSUE THREAD

**Issue by jwoo0122** [reactjs/reactjs.org#3449]
https://github.com/gatsbyjs/gatsby/pull/27615

They reverted the official support because ecosystem is not ready for that feature. If you need to use this new JSX transform feature in lastest Gatsby (in this time 2.28) you should manually setup babel.

**Comment by eps1lon** [comment]
As far as I can tell they just changed the default. They still support it but it's now opt-in:

> Disable automatic react runtime option by default.

**Comment by jwoo0122** [comment]
I think the original description maybe confused since it's not enough to use new JSX transform in gatsby by only updating to gatsby@2.24.5 or later.

**Comment by eps1lon** [comment]
I agree that the documentation should be updated to reflect the recent change. But you said they removed the support entirely which is not evident from the PR you linked. That PR only said they changed the default. It sounds like it's still supported but opt-in. Is this interpretation correct?

**Comment by jwoo0122** [comment]
Yes, I misundertood the change on Gatsby. You're right. Thank you for your correction! :) So gatsby 'support' the transform, but not use it by 'default' (opt in).

**Comment by Bartmr** [comment]
For anyone planning on turning on the new JSX Transform on Gatsby anyway, just add this to your gatsby-node.js and you should be fine

```javascript
exports.onCreateBabelConfig = ({ actions }) => {
  actions.setBabelPlugin({
    name: '@babel/plugin-transform-react-jsx',
    options: {
      runtime: 'automatic',
    },
  });
};
```

**Comment by igneosaur** [comment]
Update (Gatsby 4). To enable, in `gatsby-config.js`:

```
module.exports = {
  jsxRuntime: "automatic"
};
```

**Comment by SylvainGibert** [comment]
@igneosaur doesn't work for me :(
